pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -381,3 +390,45 @@ def register_plugins(cls, plugin_registry: Dict) -> None:
             description="DDP Plugin with `find_unused_parameters` as False",
             find_unused_parameters=False
         )
+
+    def _share_information_to_prevent_deadlock(self):
+        self._share_pids()
+
+        # remove `PL_DDP_SYNC_TMPDIR` from os.environ
+        self._sync_dir = os.environ.pop("PL_DDP_SYNC_TMPDIR", None)
+
+    def _share_pids(self):
+        """
+        Make all DDP processes aware of all processes pids.
+        """
+        self.barrier()
+        pids = self.all_gather(torch.tensor(os.getpid(), device=self.root_device))
+        pids = pids.cpu().numpy().tolist()
+        self._pids = pids if isinstance(pids, list) else [pids]
+
+    def reconciliate_processes(self, trace: str):
+        if self.world_size > 1:
+            sync_dir = self._sync_dir
+
+            if not os.path.exists(sync_dir):
+                # avoid race condition
+                try:
+                    os.makedirs(sync_dir)
+                except FileExistsError:
+                    pass
+
+            # save a file locally.
+            torch.save(trace, os.path.join(sync_dir, f"{self.global_rank}.pl"))
+
+            # sleep for a short time
+            time.sleep(3)
+
```

Comment:
you are waiting until all processese have written the file, correct?
so it is a sort of timeout. would it make sense to make this a parameter to the function with a default value?
how did you determine the value 3?

Reply:
It was already working with `1`. Just added a 3 x rule for security. But there is no special reason there.

Reply:
Done !